volumina/pixelpipeline/imagesources.py

```diff
@@ -420,8 +423,95 @@ def toImage( self ):
                 #FIXME: maybe this should be done in a better way using an operator before the colortable request which properly handles 
                 #this problem 
                 warnings.warn("Data for colortable layers cannot be float, casting",RuntimeWarning)
-                a=a.astype(np.int32)
-            vigra.colors.applyColortable(a, self._colorTable, byte_view(img))
+                a = np.asanyarray(a, dtype=np.uint32)
+
+            # If we have a masked array with a non-trivial mask, ensure that mask is made transparent.
+            _colorTable = self._colorTable
+            if np.ma.is_masked(a):
```

Comment:
Is that the right approach?  Some layers do not need (and should not have) a transparent layer.  If the color table is modified, the layer data would need to change to compensate for that.

But to answer your question: general utility functions that don't rely on other parts of volumina can go in the `volumina.utility` package.  If the function you have in mind somehow relies on functionality in `layer.py` or `imagesources.py`, then put it in one of those two files.